Source/monster.cpp

```diff
@@ -1067,24 +1067,19 @@ void MonsterHitMonster(Monster &monster, int i, int dam)
 	HitMonster(monster, dam);
 }
 
-void StartDeathFromMonster(int i, int mid)
+void StartDeathFromMonster(Monster &killerMonster, Monster &killedMonster)
 {
-	assert(static_cast<size_t>(i) < MaxMonsters);
-	Monster &killer = Monsters[i];
-	assert(static_cast<size_t>(mid) < MaxMonsters);
-	Monster &monster = Monsters[mid];
-
-	delta_kill_monster(monster, monster.position.tile, *MyPlayer);
-	NetSendCmdLocParam1(false, CMD_MONSTDEATH, monster.position.tile, mid);
+	delta_kill_monster(killedMonster, killedMonster.position.tile, *MyPlayer);
+	NetSendCmdLocParam1(false, CMD_MONSTDEATH, killedMonster.position.tile, killedMonster.getId());
 
-	if (killer.type().type == MT_GOLEM)
-		monster.whoHit |= 1 << i;
+	if (killerMonster.type().type == MT_GOLEM)
+		killedMonster.whoHit |= 1 << killerMonster.getId();

```

Comment:
This is the owning player ID, while the monster network ID is equivalent in the current code it would be good to differentiate this from the concept of a monster network ID

Reply:
Shall I create a separate issue for this? I agree with the comment but I think that changing that is outside the scope of the PR as the getId() is already being used as the network id in order places now

Reply:
This specific instance is different to other uses of Monster::getId. At least a comment calling out the difference would be good. It'll also need to be replicated in MonsterHitMonster as these two functions are the only instances where a player is given credit for a killing blow from their golem.

Reply:
Understood. I will update the PR

Reply:
I added getNetworkId, although I am not sure I understood 100% what you meant. Should calls like this all be updated to use `getNetworkId()`? 

```c++
NetSendCmdLocParam1(false, CMD_MONSTDEATH, target.position.tile, target.getId());
```

Reply:
Lets just revert this back to `target.whoHit |= 1 << attacker.getId(); // really the id the player who controls this golem`

Can add helpers to make the use stand out later.

Reply:
Thanks, I updated it

Reply:
the comments solves the misgivings i had :+1: